internal/storage/storage.go

```diff
@@ -0,0 +1,427 @@
+// Package storage holds all cached token information for MSAL. This storage can be
+// augmented with third-party extensions to provide persistent storage. In that case,
+// reads and writes in upper packages will call Serialize() to take the entire in-memory
+// representation and write it to storage and Deserialize() to update the entire in-memory
+// storage with what was in the persistent storage.  The persistent storage can only be
+// accessed in this way because multiple MSAL clients written in multiple languages can
+// access the same storage and must adhere to the same method that was defined
+// previously.
+package storage
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/AzureAD/microsoft-authentication-library-for-go/internal/json"
+	"github.com/AzureAD/microsoft-authentication-library-for-go/internal/msalbase"
+	"github.com/AzureAD/microsoft-authentication-library-for-go/internal/requests"
+)
+
+// TODO(someone): This thing does not expire tokens and DeleteCachedRefreshToken
+// is not implemented (or used anywhere).  Seems bad.
```

Comment:
It does not look like RemoveAccounts has been implemented. The token cache should have at least two public APIs: 
- For seeing what accounts and associated tokens are in the cache (`GetAllAccounts() (Account, err)`) 
- For removing accounts and associated tokens (`RemoveAccount(Account account)`) 

The implementation for RemoveAccount seems to be DeleteAccount, but I think this needs to be updated to:
- Use RemoveAccount to be consistent with other libraries
- Update paramaters to get rid of envAliases[]. User should not have to pass this in. Might also consider if we should use an Account object instead of homeAccountId. 
- RemoveAccount should remove the account and any associated accessTokens, refreshTokens, and IdTokens. In it's current implementation, I think it's only removing accounts.
- Expose RemoveAccount on clientApplication. 

I opened up an issue to track this work. In the meantime, I think DeleteCacheRefreshToken can probably be removed. #93